spaghetto/src/Spago/Config.purs

```diff
@@ -248,78 +307,61 @@ instance Semigroup Dependencies where
         Nothing, Nothing -> Nothing
         Just r, Nothing -> Just r
         Nothing, Just r -> Just r
-        Just r1, Just r2 -> Version.intersect r1 r2
+        Just r1, Just r2 -> Range.intersect r1 r2
     )
     d1
     d2
 
 instance Monoid Dependencies where
   mempty = Dependencies (Map.empty)
 
-instance Show Dependencies where
-  show (Dependencies ds) = show ds
-
-instance ToYaml Dependencies where
-  encode (Dependencies deps) = Core.fromArray
-    $ Array.fromFoldable
-    $ Map.mapMaybeWithKey
-        ( \name maybeRange -> Just $ case maybeRange of
-            Nothing -> Core.fromString (PackageName.print name)
-            Just range -> Core.jsonSingletonObject (PackageName.print name) (Core.fromString (Registry.Version.printRange range))
-        )
-        deps
-  decode =
-    let
-      decodePkgWithRange :: Object Core.Json -> Either String (Tuple PackageName (Maybe Range))
-      decodePkgWithRange obj = do
-        o <- traverse Yaml.decode obj
-        let maybeTuple = Object.toAscUnfoldable o
-        case maybeTuple of
-          Nothing -> Left "Expected Object here"
-          Just (Tuple rawPkg rawRange) -> do
-            pkgName <- lmap Parsing.parseErrorMessage $ PackageName.parse rawPkg
-            range <- lmap Parsing.parseErrorMessage $ Registry.Version.parseRange Registry.Version.Lenient rawRange
-            Right (Tuple pkgName (Just range))
-
-      decodePkg :: String -> Either String (Tuple PackageName (Maybe Range))
-      decodePkg str = case PackageName.parse str of
-        Left e -> Left (Parsing.parseErrorMessage e)
-        Right p -> Right (Tuple p Nothing)
-    in
-      Core.caseJsonArray (Left "Expected Array of Dependencies")
-        ( \arrayOfJson -> map
-            (Dependencies <<< Map.fromFoldable)
-            ( for arrayOfJson \el ->
-                (Core.caseJsonString (Left "Expected String Dependency") decodePkg el)
-                  <|> (Core.caseJsonObject (Left "Expected Object Dependency") decodePkgWithRange el)
-            )
-        )
-
-instance RegistryJson.RegistryJson Dependencies where
-  encode = Yaml.encode
-  decode = Yaml.decode
+dependenciesCodec :: JsonCodec Dependencies
+dependenciesCodec = Profunctor.dimap to from $ CA.array dependencyCodec
+  where
+  packageSingletonCodec = Internal.Codec.packageMap Range.codec
+
+  to :: Dependencies -> Array (Either PackageName (Map PackageName Range))
+  to (Dependencies deps) =
+    map
+      ( \(Tuple name maybeRange) -> case maybeRange of
+          Nothing -> Left name
+          Just r -> Right (Map.singleton name r)
+      )
+      $ Map.toUnfoldable deps :: Array _
+
+  from :: Array (Either PackageName (Map PackageName Range)) -> Dependencies
+  from = Dependencies <<< Map.fromFoldable <<< map
+    ( case _ of
+        Left name -> Tuple name Nothing
+        Right m -> rmap Just $ unsafeFromJust (List.head (Map.toUnfoldable m))
```

Comment:
```suggestion
        Right m -> map Just $ unsafeFromJust (List.head (Map.toUnfoldable m))
```

Reply:
I always considered the Functor instance of Tuple a footgun, and prefer to use the more explicit `rmap`